modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptAngularClientCodegen.java

```diff
@@ -394,6 +397,25 @@ private String applyLocalTypeMapping(String type) {
         return type;
     }
 
+    @Override
+    protected void handleMethodResponse(final Operation operation, final Map<String, Schema> schemas, final CodegenOperation op, final ApiResponse methodResponse, final Map<String, String> importMappings) {
+        super.handleMethodResponse(operation, schemas, op, methodResponse, importMappings);
+
+        if (op.responses != null && op.responses.size() > 1) {
+            String returnType = "";
```

Comment:
nit: please simplify the code by using an `ArrayList<String> returnTypes` and then joining it with `String.join(" | ", returnTypes)`